### PR TITLE
[NAE-1678] User impersonation

### PR DESCRIPTION
**Pull request by timbez**
# Description

Implements some fixes for Impersonation feature

Fixes [NAE-1678]

## Dependencies

- none

### Third party dependencies

- none

### Blocking Pull requests

EventLog classes are not resolved correctly after search in MongoDB collection (https://github.com/netgrif/application-engine/pull/108/)

## How Has Been This Tested?

Manual testing and via ImpersonationServiceTest class (src/test/groovy/com/netgrif/application/engine/impersonation/ImpersonationServiceTest.groovy)

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |       Win 10    |
| Runtime             |     Java 11      |
| Dependency Manager  |     Maven 3.8.4      |
| Framework version   |      Spring Boot 2.6.2     |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @machacjozef 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [x] I have checked my contribution with code analysis tools:
    - [x] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [x] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [ ] I have made corresponding changes to the documentation:
    - [ ] Developer documentation
    - [ ] User Guides
    - [ ] Migration Guides

[NAE-1678]: https://netgrif.atlassian.net/browse/NAE-1678?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by minop — APPROVED** [review]

**Review by renczesstefan — DISMISSED** [review]

File: src/main/java/com/netgrif/application/engine/history/domain/caseevents/DeleteCaseEventLog.java [hunk 1]
```diff
@@ -5,11 +5,15 @@
 import lombok.EqualsAndHashCode;
 import org.springframework.data.mongodb.core.mapping.Document;
 
+@Document(collection = "eventLogs")
```
Comment (renczesstefan) [inline src/main/java/com/netgrif/application/engine/history/domain/caseevents/DeleteCaseEventLog.java:8]: This may be a breaking change and may requires migration. Is this change necessary?

**Review by mazarijuraj — APPROVED** [review]

File: src/main/java/com/netgrif/application/engine/history/domain/caseevents/DeleteCaseEventLog.java (hunk 1 above)
Comment (mazarijuraj) [reply src/main/java/com/netgrif/application/engine/history/domain/caseevents/DeleteCaseEventLog.java:8]: This should not be breaking change because there is only one collection `eventLogs` right now.

**Review by mazarijuraj — APPROVED** [review]

File: src/main/java/com/netgrif/application/engine/history/domain/impersonationevents/ImpersonationEventLog.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.netgrif.application.engine.history.domain.impersonationevents;
+
+import com.netgrif.application.engine.history.domain.baseevent.EventLog;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@EqualsAndHashCode(callSuper = true)
+public class ImpersonationEventLog extends EventLog {
```
[the remark this answers is not in the document]
Comment (mazarijuraj) [reply src/main/java/com/netgrif/application/engine/history/domain/impersonationevents/ImpersonationEventLog.java]: done

**Review by mazarijuraj — APPROVED** [review]
